pkg/dns/vips/persistence.go

```diff
@@ -60,11 +62,25 @@ func (m *Persistence) GetByMesh(ctx context.Context, mesh string) (*VirtualOutbo
 		return NewEmptyVirtualOutboundView(), nil
 	}
 
-	res := map[HostnameEntry]VirtualOutbound{}
-	if err := json.Unmarshal([]byte(resource.Spec.GetConfig()), &res); err != nil {
-		return nil, err
+	var virtualOutboundView *VirtualOutboundMeshView
+	if m.useTagFirstView {
+		res := NewEmptyTagFirstOutboundView()
+		if err := json.Unmarshal([]byte(resource.Spec.GetConfig()), &res); err != nil {
+			return nil, err
+		}
+		virtualOutboundView = res.ToVirtualOutboundView()
+	} else {
+		res := map[HostnameEntry]VirtualOutbound{}
+		if err := json.Unmarshal([]byte(resource.Spec.GetConfig()), &res); err != nil {
+			return nil, err
+		}
+		virtualOutboundView, err = NewVirtualOutboundView(res)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
If we're using TagFirstView shouldn't we fallback to reading the old version for future migration to this format?
What we usually do with these changes is to first ship the flag as disable and 2 versions later we make it enabled by default. This way folks transparently adopted the new format.

Reply:
Actually feels like the deserialization should just not take into account this feature flag and just try both (because if you rollback you might have the flag off but the persisted version is the old format).